### DISCUSSION TITLE
Strange behaviour of old (error) deployments

### DISCUSSION THREAD

**Discussion by simunic-cz** [vercel/vercel#8722] [Help] [answered]
I didn't want to directly call it a deceptive practise yet, so let's call it a „strange behaviour“. I have started the 14-day free trial to dot some i's and cross some t's on my website, before I switch to the Pro plan. Out from a start I had some bugs related to combination of `next-i18next` and `next-auth`, which caused an infinite loop of error requests - I suspect from Vercel's own preview of the page (since I didn't have the site opened). I very quickly fixed the code error and made new deployments. All was well and I thought that was that. But the very next day (yesterday) my account was blocked due to „usage exceeds the fair use policy“. I was baffled since according to the Usage page, there were millions of requests made - long after I fixed the code error. After some investigating I came to the conclusion, that the most likely cause is that old deployments remain active, even after a new deployments are made and you need to manually delete old deployments.
This is really a borderline behaviour, because how on earth am I supposed to know that even if I fix a source code error that caused a loop of error requests, Vercel is still going to drain my limits due to this error on old deployments?
I had a bug - it can happen. But leaving old deployments active and draining my limits with these old buggy deployments - Vercel, that's on you. So, thank you very much...
I'm too deep now and don't have a lot of time, but as soon as I can, I'm gonna consider switching to Netlify.

**Comment by amyegan** [comment, marked as answer]
Oh no! 😞 I'm so sorry you ran into this problem @patrik-simunic-cz. Thank you for reaching out! I'm asking the team for help with this now.

**Reply by simunic-cz** [reply]
It's alright now, I guess.. It's just... you might want to consider automatically disabling old deployments? Or maybe just disabling the page preview (at least if the preview load fails a certain number of times)?
Maybe not even on production, but definitely on non-production environments. Because bugs happen and always will - and that precisely why non-prod environments exists. So if a bug in non-prod env causes an issue even after it's fixed in new deployment, that's an issue in of its self..

**Reply by simunic-cz** [reply]
Btw. I may have overreacted a bit, sorry.. I'm a lazy kind of developer and otherwise like Vercel, so I'm definitely staying... 😄

**Reply by amyegan** [reply]
I asked around and learned a few things about this problem. 

### Quick solution
If you have not already, please [delete all error loop deployments](https://vercel.com/guides/how-do-i-delete-an-individual-deployment). Also make sure you have set `NEXTAUTH_SECRET` as an [environment variable](https://vercel.com/guides/how-to-add-vercel-environment-variables) for the project. Those two things will stop the error loop from eating up your usage limit.

After you confirm that all old error loop deployments are deleted and the environment variable is set, we can unblock your account.

### More detail
Old deployments do remain active. Unwanted, misbehaving deployments should be deleted as they are not automatically removed or overwritten at this time (though some related features might already be under consideration for future development 😉 )

I imagine you already figured out a solution, since you mentioned fixing the code error, but in case anyone else comes across this error I'm sharing some detail about it here. The error you saw seems to be from [a next-auth bug](https://github.com/nextauthjs/next-auth/pull/5442) that was recently fixed. It was getting stuck in a loop because of a missing env var. The behavior was seen on other platforms and local environments as well. If `NEXTAUTH_SECRET` is missing for the deployment environment then it will get stuck in an error loop. If you're using [`next-auth@4.12.2`](https://github.com/nextauthjs/next-auth/releases/tag/next-auth%40v4.12.2), this particular error loop should no longer be happening. Please let us know if you're still seeing it with v4.12.2+ 🙏

**Reply by simunic-cz** [reply]
@amyegan Thank you 😊 Yes, I've already fixed both bugs in code (`next-i18next`), set the appropriate environment variables (`next-auth`) and deleted old deployments.
Otherwise my account is already unblocked as I needed to debug some things online so I had to switch to Pro plan.
